pkg/ccl/backupccl/backupinfo/backupinfo_test: TestMetadataSST failed

pkg/ccl/backupccl/backupinfo/backupinfo_test.TestMetadataSST [failed](https://tanzanite.cluster.engflow.com/invocations/default/9e314f26-e389-4c63-9066-d75bf5b1bb33?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbC9iYWNrdXBpbmZvOmJhY2t1cGluZm9fdGVzdA==) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
        	            	-  RowCount: (uint64) 1,
        	            	-  DistinctCount: (uint64) 1,
        	            	-  NullCount: (uint64) 0,
        	            	-  HistogramData: (*stats.HistogramData)({
        	            	-   ColumnType: (*types.T)({
        	            	-    InternalType: (types.InternalType) {
        	            	-     Family: (types.Family) 1,
        	            	-     Width: (int32) 64,
        	            	-     Precision: (int32) 0,
        	            	-     ArrayDimensions: ([]int32) <nil>,
        	            	-     Locale: (*string)(""),
        	            	-     VisibleType: (int32) 0,
        	            	-     ArrayElemType: (*types.Family)(<nil>),
        	            	-     TupleContents: ([]*types.T) <nil>,
        	            	-     TupleLabels: ([]string) <nil>,
        	            	-     Oid: (oid.Oid) 20,
        	            	-     ArrayContents: (*types.T)(<nil>),
        	            	-     TimePrecisionIsSet: (bool) false,
        	            	-     IntervalDurationField: (*types.IntervalDurationField)(<nil>),
        	            	-     GeoMetadata: (*types.GeoMetadata)(<nil>),
        	            	-     UDTMetadata: (*types.PersistentUserDefinedTypeMetadata)(<nil>)
        	            	-    },
        	            	-    TypeMeta: (types.UserDefinedTypeMetadata) {
        	            	-     Name: (*types.UserDefinedTypeName)(<nil>),
        	            	-     EnumData: (*types.EnumMetadata)(<nil>),
        	            	-     Version: (uint32) 0,
        	            	-     ImplicitRecordType: (bool) false
        	            	-    }
        	            	-   }),
        	            	-   Buckets: ([]stats.HistogramData_Bucket) (len=1) {
        	            	-    (stats.HistogramData_Bucket) {
        	            	-     NumEq: (int64) 1,
        	            	-     NumRange: (int64) 0,
        	            	-     DistinctRange: (float64) 0,
        	            	-     UpperBound: ([]uint8) (len=1) {
        	            	-      00000000  88                                                |.|
        	            	-     }
        	            	-    }
        	            	-   },
        	            	-   Version: (stats.HistogramVersion) 2
        	            	-  }),
        	            	-  AvgSize: (uint64) 1,
        	            	-  PartialPredicate: (string) "",
        	            	-  FullStatisticID: (uint64) 0
        	            	  })
        	Test:       	TestMetadataSST
    testutils.go:289: no Invalid Descriptors
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestMetadataSST2848737534
--- FAIL: TestMetadataSST (89.88s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetadataSST.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34940

pkg/ccl/backupccl/backupinfo/backupinfo_test.TestMetadataSST [failed](https://tanzanite.cluster.engflow.com/invocations/default/3c84960c-5927-477f-881f-5c6154b5b97d?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbC9iYWNrdXBpbmZvOmJhY2t1cGluZm9fdGVzdA==) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
        	            	-  RowCount: (uint64) 1,
        	            	-  DistinctCount: (uint64) 1,
        	            	-  NullCount: (uint64) 0,
        	            	-  HistogramData: (*stats.HistogramData)({
        	            	-   ColumnType: (*types.T)({
        	            	-    InternalType: (types.InternalType) {
        	            	-     Family: (types.Family) 1,
        	            	-     Width: (int32) 64,
        	            	-     Precision: (int32) 0,
        	            	-     ArrayDimensions: ([]int32) <nil>,
        	            	-     Locale: (*string)(""),
        	            	-     VisibleType: (int32) 0,
        	            	-     ArrayElemType: (*types.Family)(<nil>),
        	            	-     TupleContents: ([]*types.T) <nil>,
        	            	-     TupleLabels: ([]string) <nil>,
        	            	-     Oid: (oid.Oid) 20,
        	            	-     ArrayContents: (*types.T)(<nil>),
        	            	-     TimePrecisionIsSet: (bool) false,
        	            	-     IntervalDurationField: (*types.IntervalDurationField)(<nil>),
        	            	-     GeoMetadata: (*types.GeoMetadata)(<nil>),
        	            	-     UDTMetadata: (*types.PersistentUserDefinedTypeMetadata)(<nil>)
        	            	-    },
        	            	-    TypeMeta: (types.UserDefinedTypeMetadata) {
        	            	-     Name: (*types.UserDefinedTypeName)(<nil>),
        	            	-     EnumData: (*types.EnumMetadata)(<nil>),
        	            	-     Version: (uint32) 0,
        	            	-     ImplicitRecordType: (bool) false
        	            	-    }
        	            	-   }),
        	            	-   Buckets: ([]stats.HistogramData_Bucket) (len=1) {
        	            	-    (stats.HistogramData_Bucket) {
        	            	-     NumEq: (int64) 1,
        	            	-     NumRange: (int64) 0,
        	            	-     DistinctRange: (float64) 0,
        	            	-     UpperBound: ([]uint8) (len=1) {
        	            	-      00000000  88                                                |.|
        	            	-     }
        	            	-    }
        	            	-   },
        	            	-   Version: (stats.HistogramVersion) 2
        	            	-  }),
        	            	-  AvgSize: (uint64) 1,
        	            	-  PartialPredicate: (string) "",
        	            	-  FullStatisticID: (uint64) 0
        	            	  })
        	Test:       	TestMetadataSST
    testutils.go:289: no Invalid Descriptors
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestMetadataSST2336022951
--- FAIL: TestMetadataSST (133.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetadataSST.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/backupccl/backupinfo/backupinfo_test.TestMetadataSST [failed](https://tanzanite.cluster.engflow.com/invocations/default/3248e418-de91-40d2-871c-3359f9d01e35?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbC9iYWNrdXBpbmZvOmJhY2t1cGluZm9fdGVzdA==) on master @ [4f31fbae529e459c00ef2490e193c0c246e8d0f9](https://github.com/cockroachdb/cockroach/commits/4f31fbae529e459c00ef2490e193c0c246e8d0f9):


```
        	            	-  RowCount: (uint64) 1,
        	            	-  DistinctCount: (uint64) 1,
        	            	-  NullCount: (uint64) 0,
        	            	-  HistogramData: (*stats.HistogramData)({
        	            	-   ColumnType: (*types.T)({
        	            	-    InternalType: (types.InternalType) {
        	            	-     Family: (types.Family) 1,
        	            	-     Width: (int32) 64,
        	            	-     Precision: (int32) 0,
        	            	-     ArrayDimensions: ([]int32) <nil>,
        	            	-     Locale: (*string)(""),
        	            	-     VisibleType: (int32) 0,
        	            	-     ArrayElemType: (*types.Family)(<nil>),
        	            	-     TupleContents: ([]*types.T) <nil>,
        	            	-     TupleLabels: ([]string) <nil>,
        	            	-     Oid: (oid.Oid) 20,
        	            	-     ArrayContents: (*types.T)(<nil>),
        	            	-     TimePrecisionIsSet: (bool) false,
        	            	-     IntervalDurationField: (*types.IntervalDurationField)(<nil>),
        	            	-     GeoMetadata: (*types.GeoMetadata)(<nil>),
        	            	-     UDTMetadata: (*types.PersistentUserDefinedTypeMetadata)(<nil>)
        	            	-    },
        	            	-    TypeMeta: (types.UserDefinedTypeMetadata) {
        	            	-     Name: (*types.UserDefinedTypeName)(<nil>),
        	            	-     EnumData: (*types.EnumMetadata)(<nil>),
        	            	-     Version: (uint32) 0,
        	            	-     ImplicitRecordType: (bool) false
        	            	-    }
        	            	-   }),
        	            	-   Buckets: ([]stats.HistogramData_Bucket) (len=1) {
        	            	-    (stats.HistogramData_Bucket) {
        	            	-     NumEq: (int64) 1,
        	            	-     NumRange: (int64) 0,
        	            	-     DistinctRange: (float64) 0,
        	            	-     UpperBound: ([]uint8) (len=1) {
        	            	-      00000000  88                                                |.|
        	            	-     }
        	            	-    }
        	            	-   },
        	            	-   Version: (stats.HistogramVersion) 2
        	            	-  }),
        	            	-  AvgSize: (uint64) 1,
        	            	-  PartialPredicate: (string) "",
        	            	-  FullStatisticID: (uint64) 0
        	            	  })
        	Test:       	TestMetadataSST
    testutils.go:289: no Invalid Descriptors
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestMetadataSST2183231535
--- FAIL: TestMetadataSST (98.45s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetadataSST.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/backupinfo/backup_metadata_test.go
+++ b/pkg/ccl/backupccl/backupinfo/backup_metadata_test.go
@@ -285,6 +285,11 @@ func checkStats(
 		expectedStats = nil
 	}
 
+	sort.Slice(expectedStats, func(i, j int) bool {
+		return expectedStats[i].TableID < expectedStats[j].TableID ||
+			(expectedStats[i].TableID == expectedStats[j].TableID && expectedStats[i].StatisticID < expectedStats[j].StatisticID)
+	})
+
 	it := bm.NewStatsIter(ctx)
 	defer it.Close()
 	metaStats, err := bulk.CollectToSlice(it)

